### ISSUE TITLE
`Tarball is not in network and can not be located in cache`, with a certain combination of yarn.lock and yarn calls

### ISSUE THREAD

**Issue by abourget** [yarnpkg/yarn#2861]
https://github.com/yarnpkg/yarn/issues/560 is closed, so following up here:

Ok, got reproducible steps here, in a clean dir put:

`Dockerfile`:
```
FROM node:6.9
ADD . /app/
RUN wget https://github.com/yarnpkg/yarn/releases/download/v0.20.4/yarn-0.20.4.js -O /usr/local/bin/yarn && chmod +x /usr/local/bin/yarn
RUN cd /app && yarn
CMD /bin/bash
```

in `yarn.lock`:
```
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


bluebird@2.9.6:
  version "2.9.6"
  resolved bluebird-2.9.6.tgz#1fc3a6b1685267dc121b5ec89b32ce069d81ab7d

bluebird@^2.9.33:
  version "2.11.0"
  resolved bluebird-2.11.0.tgz#534b9033c022c9579c56ba3b3e5a5caafbb650e1

bluebird@^3.4.1:
  version "3.4.7"
  resolved bluebird-3.4.7.tgz#f72d760be09b7f76d08ed8fae98b289a8d05fab3
```

`package.json`:
```
{
  "name": "hello",
  "description": "world",
  "private": true,
  "dependencies": {
    "babyparse": "0.4.2",
    "bluebird": "2.9.6"
  }
}
```

Run `docker build -t ohmy .`

and debug :)

Final output looks like:
```
...
Step 4 : RUN cd /app && yarn
 ---> Running in 6ae3fd9a9f2c
yarn install v0.20.4
[1/4] Resolving packages...
[2/4] Fetching packages...
error "bluebird-2.9.6.tgz": Tarball is not in network and can not be located in cache ("/app/bluebird-2.9.6.tgz")
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
The command '/bin/sh -c cd /app && yarn' returned a non-zero code: 1
```

**Comment by marvinhagemeister** [comment]
Seems to be related to https://github.com/yarnpkg/yarn/issues/2920

**Comment by arcanis** [comment]
Yeah - you were using an offline cache, but probably didn't passed it to your docker instance, so it wasn't able to find the tarballs. This should be fixed in the latest release (it will fallback on the network), but you should still make sure you don't have an unwanted offline mirror somewhere (check your `.yarnrc` files).

**Comment by asfernandes** [comment]
In my case I have a middleware (Nexus) mirror. Why this isn't a supported usage of yarn?

**Comment by arcanis** [comment]
Not sure we're talking about the same kind of mirror. I'm talking about the [offline mirror](https://yarnpkg.com/blog/2016/11/24/offline-mirror/), a Yarn feature that allows you to store the tarball of your packages inside a directory (so that they can be fetched even when offline). It seems you've (maybe accidentally) enabled this feature, but didn't mount the directory into your docker instance. Yarn being unable to locate the tarball, the installation failed.

**Comment by asfernandes** [comment]
Message says "Tarball is not in *network* and can not be located in cache". If it's not in cache, it should be located in network. In my problem, when I delete the lock file it locates (and changes the lock file). In my understand, a offline mirror is a cache and same lock entries used for the cache should be valid and locate them in the internet when not in cache.

**Comment by arcanis** [comment]
The offline mirror and the cache are two separate things. If the offline mirror cannot be found, the installation will not proceed in the older Yarn releases. Starting from the 0.23 release, this behavior changed, and we will fallback on the network if the offline mirror cannot be found.

**Comment by troywweber7** [comment]
So what is the agreed upon solution to this? I'm on `yarn 1.22.4`.

**Comment by andreialecu** [comment]
If anyone is running into this with `yarn workspaces` and local tarballs, try running `yarn install` from the **root of the project**, not from within a workspace directory (make sure the current directory is the root of the project).

The bug is on this line: https://github.com/yarnpkg/yarn/blob/0e7133ca28618513503b4e1d9063f1c18ea318e5/src/fetchers/tarball-fetcher.js#L214

If you're `cd`'d to any other folder than the root, it will resolve to the wrong path.

**Comment by seanlaff** [comment]
I ran into this, but it was a silly mistake of having a docker build where I copied in `package.json`/`yarn.lock` as a stage, ran install, and then copied the rest of my project in. I needed to (obviously) copy in the tar file before running install.

**Comment by birgersp** [comment]
The error occured for me when I ran `yarn` in a workspace folder (not the workspace root). After removing node_modules and clearing yarn cache I could run `yarn` in the workspace root and all was well. We also ran into a issue where the ts compiler still couldn't find the dependency, even though yarn install was fine. This happened because tsc had created the `tsconfig.tsbuildinfo` file when the dependencies were wrong so we had to delete tsconfig.tsbuildinfo as well. Just a note for any potential future frustrations with this ;)
